libpod/image/image.go

```diff
@@ -18,63 +30,112 @@ type Image struct {
 	inspect.ImageData
 	InputName string
 	Local     bool
-	runtime   *libpod.Runtime
-	image     *storage.Image
+	//runtime   *libpod.Runtime
+	image        *storage.Image
+	imageruntime *ImageRuntime
+}
+
+type ImageRuntime struct {
+	store               storage.Store
+	signaturePolicyPath string
+	authfile            string
+	writer              io.Writer
+	dockeroptions       *DockerRegistryOptions
+	signingoptions      SigningOptions
+}
+
+func NewImageRuntime(options storage.StoreOptions) (*ImageRuntime, error) {
+	if reexec.Init() {
+		return nil, errors.Errorf("dude")
+	}
+	store, err := setStore(options)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ImageRuntime{
+		store: store,
+	}, nil
+}
+
+func setStore(options storage.StoreOptions) (storage.Store, error) {
+	store, err := storage.GetStore(options)
+	if err != nil {
+		return nil, err
+	}
+	is.Transport.SetStore(store)
+	return store, nil
+}
+
+// newFromStorage creates a new image object from a storage.Image
+func (ir *ImageRuntime) newFromStorage(img *storage.Image) *Image {
+	image := Image{
+		InputName:    img.ID,
+		Local:        true,
+		imageruntime: ir,
+		image:        img,
+	}
+	return &image
 }
 
 // NewFromLocal creates a new image object that is intended
 // to only deal with local images already in the store (or
 // its aliases)
-func NewFromLocal(name string, runtime *libpod.Runtime) (Image, error) {
+func (ir *ImageRuntime) NewFromLocal(name string) (*Image, error) {
+
 	image := Image{
-		InputName: name,
-		Local:     true,
-		runtime:   runtime,
+		InputName:    name,
+		Local:        true,
+		imageruntime: ir,
 	}
 	localImage, err := image.getLocalImage()
 	if err != nil {
-		return Image{}, err
+		return nil, err
 	}
 	image.image = localImage
-	return image, nil
+	return &image, nil
 }
 
 // New creates a new image object where the image could be local
 // or remote
-func New(name string, runtime *libpod.Runtime) (Image, error) {
+func (ir *ImageRuntime) New(name, signaturePolicyPath, authfile string, writer io.Writer, dockeroptions *DockerRegistryOptions, signingoptions SigningOptions) (*Image, error) {
 	// We don't know if the image is local or not ... check local first
 	newImage := Image{
-		InputName: name,
-		Local:     false,
-		runtime:   runtime,
+		InputName:    name,
+		Local:        false,
+		imageruntime: ir,
 	}
 	localImage, err := newImage.getLocalImage()
 	if err == nil {
 		newImage.Local = true
 		newImage.image = localImage
-		return newImage, nil
+		return &newImage, nil
 	}
 
 	// The image is not local
-	pullNames, err := newImage.createNamesToPull()
+	// Assign the options provided to imageruntime
+	newImage.imageruntime.authfile = authfile
```

Comment:
Do we want to hold these in the runtime? Can we just make them arguments to Pull and pass them straight in?

Reply:
We discussed this I believe. Because New can mean local or remote and nobody should be directly calling PullImage, the args are needed up front.  As for how they are stored, we can alter that if you like.

Reply:
Because no one is directly calling PullImage, why is it bad to make these arguments to it? We're going to be calling it from here always, so we can make the signature whatever we want

Reply:
agree , removed